compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2791,11 +2791,13 @@ impl Target {
             matches!(self.linker_flavor, LinkerFlavor::Bpf),
             "`linker_flavor` must be `bpf` if and only if `arch` is `bpf`"
         );
-        check_eq!(
-            self.arch == Arch::Nvptx64,
-            matches!(self.linker_flavor, LinkerFlavor::Ptx),
-            "`linker_flavor` must be `ptc` if and only if `arch` is `nvptx64`"
-        );
+        if self.arch == Arch::Nvptx64 {
+            check_matches!(
+                self.linker_flavor,
+                LinkerFlavor::Llbc,
+                "`linker_flavor` must be `llbc` if `arch` is `nvptx64`"
+            );
+        }
```

Comment:
So, this tested before that `LinkerFlavor::Ptx` _wasn't_ set by default on any other target than `nvptx64`.

Would it make sense to keep that property somewhat? Something like:
```rust
else {
    check!(!matches!(self.linker_flavor, LinkerFlavor::Ptx), "`linker_flavor` `ptc` is not be used by default anywhere");
}
```

Reply:
I must admit I don't see the exact situation this consistency check guard against so it's hard for me to have an opinion. I did as you suggested and remove it.